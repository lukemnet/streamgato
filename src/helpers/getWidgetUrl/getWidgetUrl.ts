import constructURLparams from 'helpers/constructURLparams/constructURLparams';
import getWidgetOrigin from 'helpers/getWidgetOrigin/getWidgetOrigin';
import { ShorthandValues } from 'types';

interface GetWidgetUrlParams {
  origin: string;
  alias: string;
  params?: ShorthandValues;
}

const getWidgetUrl = ({
  origin,
  alias,
  params,
}: GetWidgetUrlParams) => {
  console.log(process.env.NODE_ENV) // eslint-disable-line
  const widgetOrigin = getWidgetOrigin(origin);
  const urlParams = params
    ? constructURLparams(params)
    : undefined;
  return `${widgetOrigin}/${alias}${urlParams}`;
};

export default getWidgetUrl;
