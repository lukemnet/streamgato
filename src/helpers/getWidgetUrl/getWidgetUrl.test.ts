import getWidgetUrl from './getWidgetUrl';

const testParams = {
  alias: 'sm',
  params: {
    h: 1,
    m: 2,
    s: 3,
  },
};

it('should construct widget URL', () => {
  expect(getWidgetUrl(testParams.alias, testParams.params)).toMatchSnapshot();
});