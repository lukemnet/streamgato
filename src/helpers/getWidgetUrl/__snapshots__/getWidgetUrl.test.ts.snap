// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should construct widget URL 1`] = `"http://localhost/sm?h=1&m=2&s=3"`;
